### PR TITLE
Describe dwell time handling for AS923

### PR DESCRIPTION
**Pull request by bastbu**
# PR for issue #732

## What is being addressed

Describes the decisions taken with respect to dwell time support in AS923.

### CODE REVIEW

**Review by Mandur — APPROVED** [review]

File: docs/adr/004_region_as923_implementation.md [hunk 1]
```diff
@@ -2,34 +2,119 @@
...
 `Channel 1 frequency Hz = 923400000 + AS923_FREQ_OFFSET_HZ`
 
 Using this formula we will calculate the offset by subtracting 923200000 from the configured channel 0 frequency. We will use the formula for channel 1 frequency to validate the offset value and throw and exception if values are not the same.
 
-In the implementation of region AS923 the frequencies for channel 0 and 1 will be passed to the region-specific constructor where the offset value will be calculated.  
+In the implementation of region AS923 the frequencies for channel 0 and 1 will be passed to the region-specific constructor where the offset value will be calculated.
+
+### Dwell Times
+
+We will support dwell times through the automatic dwell time management process described in the appendix.
```
Comment (MaggieSalak) [inline docs/adr/004_region_as923_implementation.md:50]: My understanding after yesterday's meeting was that we would implement manual dwell time management?

Comment (bastbu) [reply docs/adr/004_region_as923_implementation.md]: Based on the feedback from @p-schuler we reconsidered, changed the description of automatic dwell time management and propose to use that approach instead.

File: docs/adr/004_region_as923_implementation.md [hunk 2]
```diff
@@ -2,34 +2,118 @@
...
+
+## Appendix
+
+### Dwell times
+
+We discuss different approaches to handle dwell time limitations.
+
+#### No `TxParamSetupReq` support
+
+If we do not support the `TxParamSetupReq` MAC Command as part of the initial version of the AS923 region implementation. One implication is that the maximum Equivalent Isotropically Radiated Power (EIRP), which is normally configurable through `TxParamSetupReq`, will be static and non-configurable. We introduce a Boolean configuration value in the concentrator device twin that specifies whether a dwell time limitation applies to that concentrator. Based on this configuration value, the gateway will apply a different set of regional parameters, which take the dwell time limitations into account. It will be the responsibility of the starter kit user to ensure that end devices are configured with the same dwell time settings as the concentrator to which they are connected, as it will not be automatically propagated to the end device through `TxParamSetupReq`.
```
Comment (MaggieSalak) [inline docs/adr/004_region_as923_implementation.md:60]: 'If' seems out of place here.

File: docs/adr/004_region_as923_implementation.md [hunk 3]
```diff
@@ -2,34 +2,118 @@
...
+
+Pros:
+
+- Easy to implement
+
+Cons:
+
+- End devices (e.g. by Netvox) cannot be updated with different dwell time limitation defaults after shipping. If local regulations change, such devices cannot be used any longer with the starter kit
```
Comment (MaggieSalak) [inline docs/adr/004_region_as923_implementation.md:68]: That's a valid argument but I don't quite get how it's only applicable to this case and not the other options? Can the dwell time setting on these devices still be modified with a MAC command?

Comment (bastbu) [reply docs/adr/004_region_as923_implementation.md:68]: Yes, that can be done using `TxParamSetupReq`.

**Review by MaggieSalak — APPROVED** [review]

File: docs/adr/004_region_as923_implementation.md (hunk 1 above)
Comment (bastbu) [reply docs/adr/004_region_as923_implementation.md]: Resolving the comment as a consequence of the approval

**Pull request merged**